src/libcrun/handlers/spin.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * crun - OCI runtime written in C
+ *
+ * Copyright (C) 2017, 2018, 2019, 2020, 2021 Giuseppe Scrivano <giuseppe@scrivano.org>
+ * crun is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * crun is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with crun.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#define _GNU_SOURCE
+
+#include "../custom-handler.h"
+#include "../linux.h"
+#include <sys/stat.h>
+
+
+#ifdef HAVE_DLOPEN
+#  include <dlfcn.h>
+#endif
+
+#ifdef HAVE_SPIN
+
+#endif
+
+#if HAVE_DLOPEN && HAVE_SPIN
+static int
+spin_exec (void *cookie arg_unused, libcrun_container_t *container arg_unused,
+           const char *pathname arg_unused, char *const argv[] arg_unused)
+{
+    // wasmtime fails to determine default config path if $HOME is not set
+    char *newenviron[] = { "HOME=/root", NULL };
+    
+    char *newargv[5];
+    newargv[0] = "/bin/spin";
+    newargv[1] = "up";
+    newargv[2] = "--listen";
+    newargv[3] = "0.0.0.0:80";
+    newargv[4] = NULL;
```

Comment:
Can this be inline `char *newargv[] = {"/bin/spin", "up", ....}`

Reply:
Sure, good point!